src/SonarQube.Client/Services/SonarQubeClient.cs

```diff
@@ -35,46 +35,64 @@
 
 namespace SonarQube.Client.Services
 {
-    public class SonarQubeClient : ISonarQubeClient
+    public sealed class SonarQubeClient : ISonarQubeClient, IDisposable
     {
-        private readonly HttpMessageHandler messageHandler;
-        private readonly TimeSpan requestTimeout;
+        private readonly HttpClient httpClient;
+        private bool isDisposed;
 
-        public SonarQubeClient(HttpMessageHandler messageHandler, TimeSpan requestTimeout)
+        public SonarQubeClient(ConnectionRequest connection, HttpMessageHandler messageHandler, TimeSpan requestTimeout)
         {
+            if (connection == null)
+            {
+                throw new ArgumentNullException(nameof(connection));
+            }
             if (messageHandler == null)
             {
                 throw new ArgumentNullException(nameof(messageHandler));
             }
-
             if (requestTimeout <= TimeSpan.Zero)
             {
                 throw new ArgumentException("Doesn't expect a zero or negative timeout.", nameof(requestTimeout));
             }
 
-            this.messageHandler = messageHandler;
-            this.requestTimeout = requestTimeout;
+            this.httpClient = new HttpClient(messageHandler)
+            {
+                BaseAddress = connection.ServerUri,
+                Timeout = requestTimeout
+            };
+            this.httpClient.DefaultRequestHeaders.Authorization = AuthenticationHeaderFactory.Create(connection);
         }
 
-        public Task<Result<ComponentResponse[]>> GetComponentsSearchProjectsAsync(ConnectionRequest connection,
-                                            ComponentRequest request, CancellationToken token)
+        public void Dispose()
+        {
+            if (isDisposed)
+            {
+                return;
+            }
+
+            this.httpClient.Dispose();
+            this.isDisposed = true;
+        }
+
+        public Task<Result<ComponentResponse[]>> GetComponentsSearchProjectsAsync(ComponentRequest request,
+            CancellationToken token)
         {
             const string SearchProjectsAPI = "api/components/search_projects"; // Since 6.2; internal
 
             var query = AppendQueryString(SearchProjectsAPI, "?organization={0}&p={1}&ps={2}&asc=true",
-                        request.OrganizationKey, request.Page, request.PageSize); // TODO: should handle optional params
+                request.OrganizationKey, request.Page, request.PageSize); // TODO: should handle optional params
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this 'TODO' comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=csharpsquid%3AS1135)